p2p/p2p/src/inbound_server.rs

```diff
@@ -102,8 +114,36 @@ where
                 .instrument(Span::current()),
             );
         } else {
+            // Otherwise check if the node is simply pinging us.
             tracing::debug!("No permit free for incoming connection.");
-            // TODO: listen for if the peer is just trying to ping us to see if we are reachable.
+            
+            // We only handle 2 ping request conccurently. Otherwise we drop the connection immediately.
+            if ping_join_set.len() < 2 {
+                ping_join_set.spawn(
+                    async move {
+                        // Await first message from node. If it is a ping request we respond back, otherwise we drop the connection.
+                        let fut = timeout(Duration::from_secs(5), peer_stream.next());
```

Comment:
Also can you move the 2 & 5 here to `constants`  with appropriate names for each please